ffi/src/lib.rs

```diff
@@ -313,162 +568,365 @@ fn unwrap_conversion<T>(value: Option<T>, expected_type: ValueType) -> T {
     }
 }
 
-// as_long
+/// Consumes a `TypedValue` and returns the value as a `long`.
+///
+/// # Panics
+///
+/// If the `ValueType` of the `TypedValue` is not `ValueType::Long`.
+///
+/// # Safety
+///
+/// Callers must ensure that the pointer to the `TypedValue` is not dangling .
 #[no_mangle]
 pub unsafe extern "C" fn typed_value_as_long(typed_value: *mut TypedValue) ->  c_longlong {
     let typed_value = Box::from_raw(typed_value);
     unwrap_conversion(typed_value.into_long(), ValueType::Long)
 }
 
-// as_entid
+/// Consumes a `TypedValue` and returns the value as an `Entid`.
+///
+/// # Panics
+///
+/// If the `ValueType` of the `TypedValue` is not `ValueType::Ref`.
+///
+/// # Safety
+///
+/// Callers must ensure that the pointer to the `TypedValue` is not dangling .
 #[no_mangle]
 pub unsafe extern "C" fn typed_value_as_entid(typed_value: *mut TypedValue) ->  Entid {
     let typed_value = Box::from_raw(typed_value);
     unwrap_conversion(typed_value.into_entid(), ValueType::Ref)
 }
 
-// kw
+/// Consumes a `TypedValue` and returns the value as an keyword `String`.
+///
+/// # Panics
+///
+/// If the `ValueType` of the `TypedValue` is not `ValueType::Ref`.
+///
+/// # Safety
+///
+/// Callers must ensure that the pointer to the `TypedValue` is not dangling .
 #[no_mangle]
 pub unsafe extern "C" fn typed_value_as_kw(typed_value: *mut TypedValue) ->  *const c_char {
     let typed_value = Box::from_raw(typed_value);
     string_to_c_char(unwrap_conversion(typed_value.into_kw(), ValueType::Keyword).to_string())
 }
 
-//as_boolean
+/// Consumes a `TypedValue` and returns the value as a boolean represented as an `i32`.
+/// If the value of the boolean is `true` the value returned is 1.
+/// If the value of the boolean is `false` the value returned is 0.
+///
+/// # Panics
+///
+/// If the `ValueType` of the `TypedValue` is not `ValueType::Boolean`.
+///
+/// # Safety
+///
+/// Callers must ensure that the pointer to the `TypedValue` is not dangling .
 #[no_mangle]
 pub unsafe extern "C" fn typed_value_as_boolean(typed_value: *mut TypedValue) -> i32 {
```

Comment:
Generally "into" means "consume" and "as" means "take a reference".  This looks like it's taking ownership (consuming), since that's what `from_raw` does.  That is, it's not OK to do:
```
void* foo = some_function_that_gives_a_Rust_TypedValue();
if (typed_value_as_boolean(foo)) {
 // do something.
}
... later ...
if (!typed_value_as_boolean(foo)) {
 // do something else.
}
```
and that's not suggested by the name.  Can we either produce a reference to the internal value (I bet we can't, since some of them can't be exposed directly to the FFI as C-level references, for example, `DateTime<Utc>` probably can't) _or_ follow the Rust convention and use `into`?